lib/proxy_api/resource.rb

```diff
@@ -63,15 +63,22 @@ def parse(response)
       false
     end
 
+    def escape_path(path)
+      parsed_path = URI.parse(path)
+      query = parsed_path.query
+      path = "#{parsed_path.path}?#{CGI.escape(query)}" unless query.nil?
```

Comment:
Are you sure this will escape it properly? for example:
```
[82] pry(main)> a= URI.parse('/aa?page=5')
=> #<URI::Generic /aa?page=5>
[83] pry(main)> CGI.escape(a.query)
=> "page%3D5"
[84] pry(main)> URI.escape '/aa?page=5'
=> "/aa?page=5"

Reply:
this is what the output will be: "/aa?page%3D5"
after adding the parsed_path.path part.

Reply:
which is incorrect for query parameters, you don't want to escape the = sign